src/traces/ohlc/helpers.js

```diff
@@ -95,14 +95,65 @@ exports.makeTransform = function(traceIn, state, direction) {
 };
 
 exports.getFilterFn = function(direction) {
+    return new _getFilterFn(direction);
+};
+
+function _getFilterFn(direction) {
+    var isPrevThisDirection = null;
+    var cPrev = null;
+    var fn;
+
     switch(direction) {
         case 'increasing':
-            return function(o, c) { return o <= c; };
+            fn = function(o, c) {
+                if(o === c) {
+                    if(c > cPrev) {
+                        return true; // increasing
+                    } else if(c < cPrev) {
+                        return false; // decreasing
+                    } else {
+                        if(isPrevThisDirection === true) {
+                            return true; // determine by last candle
+                        } else if(isPrevThisDirection === false) {
+                            return false; // determine by last candle
+                        } else {
+                            return true; // If we don't have previous data, assume it was increasing
+                        }
+                    }
+                }
+                return o < c;
+            };
+            break;
 
         case 'decreasing':
-            return function(o, c) { return o > c; };
+            fn = function(o, c) {
+                if(o === c) {
+                    if(c > cPrev) {
+                        return false; // increasing
+                    } else if(c < cPrev) {
+                        return true; // decreasing
+                    } else {
+                        if(isPrevThisDirection === true) {
+                            return true; // determine by last candle
+                        } else if(isPrevThisDirection === false) {
+                            return false; // determine by last candle
+                        } else {
+                            return false; // If we don't have previous data, assume it was increasing
+                        }
+                    }
+                }
+                return o > c;
+            };
```

Comment:
@alexcjohnson I vaguely remember you mentioning on slack that you had a commit implementing your solution on a branch. If that's not case, I'll try to push something up to get this PR in this week's `1.28.0`.